### DISCUSSION TITLE
SEQ unable to show and filter on all structured data?

### DISCUSSION THREAD

**Discussion by BondByteBlaster** [datalust/seq-tickets#2060] [Help]
Hi,

I'm using Serilog in a C# service for structured logging and facing challenges with how structured data is displayed and filtered in SEQ. Although the raw JSON export of a log entry shows detailed structured data, this data isn't fully displayed in SEQ's expanded row view. Additionally, I'm unable to filter logs based on these structured properties.

Here's an example of the raw JSON export from SEQ:

`{
    "@t": "2024-01-12T15:47:11.8522997Z",
    "@mt": "MyMessage. Object guid: {@MyContract}.",
    "@m": "MyMessage. Object guid: {\"MuObject\":{\"Guid\":\"5196dcbf-e9fa-490d-3074-08dab7f7faac\",\"MappingId\":97,\"OperatingUnitId\":0,\"Data\":[{\"UniqueId\":662,\"Id\":\"20466\",\"ItemType\":\"MyApp_Item\",\"Data\":\"\\\"2024-01-12T16:47:00\\\"\",\"DataType\":\"DateTime\",\"FromDataType\":\"0\",\"IsReturnField\":false,\"ReturnFieldIdentifier\":null,\"StringFormat\":null,\"PrimaryExternalCode\":null,\"SortOrder\":0,\"ExternalCodeSortOrder\":null,\"IsList\":false,\"$type\":\"DataValue\"},{\"UniqueId\":667,\"Id\":\"20472\",\"ItemType\":\"MyApp_Item\",\"Data\":\"\\\"BC_Felmeddelande\\\"\",\"DataType\":\"String\",\"FromDataType\":\"0\",\"IsReturnField\":true,\"ReturnFieldIdentifier\":\"ErrorMessage\",\"StringFormat\":null,\"PrimaryExternalCode\":null,\"SortOrder\":0,\"ExternalCodeSortOrder\":null,\"IsList\":false,\"$type\":\"DataValue\"}],\"FilterData\":[{\"UniqueId\":0,\"Id\":\"58\",\"ItemType\":\"MyApp_SystemItem\",\"Data\":null,\"DataType\":\"JobbTimestamp\",\"FromDataType\":\"0\",\"IsReturnField\":false,\"ReturnFieldIdentifier\":null,\"StringFormat\":null,\"PrimaryExternalCode\":null,\"SortOrder\":0,\"ExternalCodeSortOrder\":null,\"IsList\":false,\"$type\":\"DataValue\"}],\"EntityNumber\":\"95\",\"EntitySurname\":\"Bobsson\",\"EntityGivenname\":\"Bob\",\"ObjectId\":\"82468\",\"Status\":\"New\",\"$type\":\"MuObject\"},\"Created\":\"2024-01-12T16:47:11.8187197+01:00\",\"TrackData\":{\"MessageGuid\":\"851a91e3-95b0-4e47-a75b-2602ad1dafef\",\"MessageMaceinGuid\":\"00000000-0000-0000-0000-000000000000\",\"ObjectId\":82468,\"ObjectGUI\":\"5196dcbf-e9fa-490d-3074-08dab7f7faac\",\"ObjectNumber\":\"22000051\",\"DataContractId\":141,\"MappingId\":97,\"MappingText\":\"Data Types\",\"InstallationId\":17,\"InstallationText\":\"Mace\",\"$type\":\"TrackMessageData\"},\"$type\":\"SendMuObject\"}.",
    "@i": "a24a2788",
    "ApplicationLocation": "C:\\MyApp5\\Odin\\CloudConnector - Mace\\",
    "MachineName": "OdisDev",
    "ProcessId": 492,
    "ServiceInstanceName": "ConnectorService1",
    "SourceContext": "MyApp.Cloud.Connector.Services.Interface.ICloudService",
    "TrackMessageData": "MyApp.Cloud.MuObjectService.External.DataContract.TrackMessageData",
    "cloudTreatmenBContract": {
        "$type": "SendMuObject",
        "Created": "2024-01-12T16:47:11.8187197+01:00",
        "MuObject": {
            "$type": "MuObject",
            "Data": [
                {
                    "$type": "DataValue",
                    "Data": "\"2024-01-12T16:47:00\"",
                    "DataType": "DateTime",
                    "ExternalCodeSortOrder": null,
                    "FromDataType": "0",
                    "Id": "20466",
                    "IsList": false,
                    "IsReturnField": false,
                    "ItemType": "MyApp_Item",
                    "PrimaryExternalCode": null,
                    "ReturnFieldIdentifier": null,
                    "SortOrder": 0,
                    "StringFormat": null,
                    "UniqueId": 662
                },
                {
                    "$type": "DataValue",
                    "Data": "\"BC_Felmeddelande\"",
                    "DataType": "String",
                    "ExternalCodeSortOrder": null,
                    "FromDataType": "0",
                    "Id": "20472",
                    "IsList": false,
                    "IsReturnField": true,
                    "ItemType": "MyApp_Item",
                    "PrimaryExternalCode": null,
                    "ReturnFieldIdentifier": "ErrorMessage",
                    "SortOrder": 0,
                    "StringFormat": null,
                    "UniqueId": 667
                }
            ],
            "FilterData": [
                {
                    "$type": "DataValue",
                    "Data": null,
                    "DataType": "JobbTimestamp",
                    "ExternalCodeSortOrder": null,
                    "FromDataType": "0",
                    "Id": "58",
                    "IsList": false,
                    "IsReturnField": false,
                    "ItemType": "MyApp_SystemItem",
                    "PrimaryExternalCode": null,
                    "ReturnFieldIdentifier": null,
                    "SortOrder": 0,
                    "StringFormat": null,
                    "UniqueId": 0
                }
            ],
            "Guid": "5196dcbf-e9fa-490d-3074-08dab7f7faac",
            "MappingId": 97,
            "OperatingUnitId": 0,
            "EntityNumber": "95",
            "EntityGivenname": "Bob",
            "EntitySurname": "Bobsson",
            "Status": "New",
            "ObjectId": "82468"
        },
        "TrackData": {
            "$type": "TrackMessageData",
            "DataContractId": 141,
            "InstallationId": 17,
            "InstallationText": "Mace",
            "MappingId": 97,
            "MappingText": "Data Types",
            "MessageMaceinGuid": "00000000-0000-0000-0000-000000000000",
            "MessageGuid": "851a91e3-95b0-4e47-a75b-2602ad1dafef",
            "ObjectGUI": "5196dcbf-e9fa-490d-3074-08dab7f7faac",
            "ObjectId": 82468,
            "ObjectNumber": "22000051"
        }
    }
}`

In the SEQ UI, the expanded row for this log entry shows only a limited view of this data:

`12 Jan 2024 16:47:11.852
 Object successfully sent to CloudMQ. Object guid: {'$type': 'SendMyObject', Created: '2024-01-12T16:47:11.8187197+01:00', Object….
ApplicationLocation
C:\MyApp5\MyApp\CloudConnector - ABC\
cloudTreatmentContract
{'$type': 'SendMyObject', Created: '2024-01-12T16:47:11.8187197+01:00', MyObject: {…}, TrackData: {…}}
MachineName
OdisDev
ProcessId
492
ServiceInstanceName
ConnectorService1
SourceContext
MyApp.Cloud.Connector.Services.Interface.ICloudService
TrackMessageData
MyApp.Cloud.MyObjectService.External.DataContract.TrackMessageData`

As you can see, much of the structured data, particularly from properties like TrackMessageData, is missing in the SEQ UI, and I'm unable to filter based on these properties.

Could you please advise on how I can make SEQ display and index these structured properties correctly, both for visibility in the expanded row view and for filtering?

Thank you for your help!

**Comment by nblumhardt** [comment]
Hi @BondByteBlaster,

I think `TrackMessageData` will be shown by expanding the object `TrackData`, which looks like it's collapsed in your view.

If that doesn't work, including a screenshot might help us narrow down the issue further.

Best regards,
Nick
